src/PrestaShopBundle/Form/Admin/Type/DisablingToggleType.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class DisablingToggleType extends SwitchType
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+
+        $resolver
+            ->setDefaults([
+                'disable_on_match' => true,
+                'matching_value' => 0,
+                'label' => false,
+                'attr' => [
+                    'container_class' => 'disabling-toggle',
+                    'data-value-type' => 'boolean',
+                ],
+                'row_attr' => [
+                    'class' => 'ps-disabling-toggle',
+                ],
+            ])
+            ->setAllowedTypes('disable_on_match', 'bool')
```

Comment:
You should add an option with the input target, this option is required and can't be left empty which forces to correctly configure it on creation

Reply:
```suggestion
            ->setRequired('toggle_target')
            ->setAllowedTypes('toggle_target', 'string')
            ->setAllowedTypes('disable_on_match', 'bool')
```